pages/[type]/[identifier]/edit.js

```diff
@@ -0,0 +1,210 @@
+import Request from "../../../libs/request"
+import EditJson from "../../../components/edit-func"
+import PropTypes from 'prop-types'
+import React from "react"
+import { useContext, useState } from "react"
+import { Button, LinkButton } from "../../../components/button"
+import { API_URL } from "../../../libs/constants"
+import { canEdit } from "../../../libs/general"
+import SessionContext from '../../../components/session-context'
+import { useRouter } from 'next/router'
+
+/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "^_.*" }]*/
+
+const updateItem = async (path, item, session) => {
+    const response = await fetch(`${API_URL}${path}`, {
+        method: "PATCH",
+        credentials: "include",
+        headers: {
+            "Content-Type": "application/json",
+            "X-CSRF-Token": session._csrft_,
+        },
+        body: JSON.stringify(item),
+    }).then(response => {
+        return response
+    }).catch(_err => {
+        return {
+            errors: [
+                {
+                    description: "Network error while saving",
+                    names: ["unknown"],
+                },
+            ],
+        }
+    })
+    return response
+}
+
+const SaveCancelControl = ({ cancelClick, saveClick, itemPath, saveEnabled }) => {
+    return (
+        <div className="flex space-x-1">
+            <LinkButton href={itemPath} navigationClick={cancelClick}>Cancel</LinkButton>
+            <Button onClick={saveClick} enabled={saveEnabled} type={ saveEnabled ? "primary" : "info" }>Save</Button>
+        </div>)
+}
+
+SaveCancelControl.propTypes = {
+    cancelClick: PropTypes.func.isRequired,
+    saveClick: PropTypes.func.isRequired,
+    itemPath: PropTypes.string.isRequired,
+    saveEnabled: PropTypes.bool.isRequired,
+}
+
+const filterItem = (item) => {
+    const {
+        ['@type']: _type,
+        ['@id']: _id,
+        ['@context']: _context,
+        ['uuid']: _uuid,
+        ['actions']: _actions,
+        ['access_keys']: _access,
+        ['schema_version']: _version,
+        ['title']: _title,
+        ...filtered } = item
+    return filtered
+}
+
+function sortedJson(obj) {
+    if (obj instanceof Array) {
+        return obj.map((value) => sortedJson(value))
+    }
+    if (obj instanceof Object) {
+        const sorted = {}
+        Object.keys(obj).sort().forEach((key) => {
+            sorted[key] = obj[key]
+        })
+        return sorted
+    }
+    return obj
+}
+
+const SavedErrors = ({ errors = [] }) => {
+    return (
+        <div>
+            <p className="text-rose-600 text-lg">Errors from Save:</p>
+            <ul className="list-disc">
+                {errors.map((error) =>
+                    <li key={error.description} className="text-rose-600 text-base">
+                        {error.description}
+                    </li>
+                )}
+            </ul>
```

Comment:
Oh this `SavedErrors` component is for if the formatting is wrong on the server, or some other random thing happens. Ideally the server will come back with an error response that has a description of went wrong and we can display it. Since it could be a formatting issue: for example if it's supposed to be  CURIE but a curie is not supplied the server responds with a 415 or 422 or something.

Reply:
I guess we could also try at some point is using the same template validation as on the backend and show those errors also? Could be a cool feature at some point.

Reply:
On validation, I’d love to see some standard way, possibly shared across back end and front end, to show (or I suppose in some cases hide) errors to the user in a consistent way. It could be a modal, or maybe a small message in some consistent area — I don’t know.